x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -289,6 +289,12 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
         logger: this.logger,
         cache: this.exceptionsCache,
       });
+
+      // initialize the global agnostic Trusted Apps list
+      exceptionListClient.createTrustedAppsList().catch((error) => {
```

Comment:
I add the creation of the list during `start` of the Security Solution plugin in order to not sprinkle through calls to the `createTrustedAppsList()`. Its a little different from what is done for Exceptions, so please let me know your thoughts.

Reply:
Again, it's super unclear how and when and where these lists should be created.

For the endpoint exception list, we currently create it when the first rule is created (or when any subsequent rule is created, if somehow the list was deleted).

We can't do that for trusted apps, since they are independent of rules. 

I realize Frank said we'd like to avoid creating resources on startup, but it's unclear to me what the best solution is, in that case. Any attempt to create the list dynamically will be racy, and somewhat unreliable (and will also result in unexpected behavior IMO).

Needs wider discussion, perhaps when @FrankHassanabad gets back. <--- @peluja1012 @spong